### PR TITLE
[WIP] Flatpak package support

### PR DESCRIPTION
**Pull request by nickavem**
### Description of the Change

Add support for flatpak and automatically publish to flathub.org

### TODO

- [x] Build initial flatpak package (https://github.com/nickavem/io.atlauncher.ATLauncher)
- [x] Receive upstream permission #401
- [ ] Publish to Flathub
- [ ] Automate with Github CI #401

### Related Issues
#401

### CODE REVIEW

File: packaging/linux/flatpak/com.atlauncher.ATLauncher.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+      buildsystem: simple
+      build-commands:
+          - install -D com.atlauncher.ATLauncher.desktop /app/share/applications/com.atlauncher.ATLauncher.desktop
+          - install -D com.atlauncher.ATLauncher.metainfo.xml /app/share/metainfo/com.atlauncher.ATLauncher.appdata.xml
+      sources:
+          - type: file
+            path: com.atlauncher.ATLauncher.metainfo.xml
+          - type: file
+            path: com.atlauncher.ATLauncher.desktop
+
```
[the remark this answers is not in the document]
Comment (nickavem) [reply packaging/linux/flatpak/com.atlauncher.ATLauncher.yml:56]: They are there to implement the desktop files. While close, the upstream metadata and .desktop don't work perfectly with flatpak and had to be modified (and are included in this PR). I could switch it to a remote fetch, if that would be easier.

**Comment by jamierocks** [comment]
> Publish to Flathub

Given their stance on software piracy - should that be a goal?

**Comment by nickavem** [comment]
> > Publish to Flathub
> 
> Given their stance on software piracy - should that be a goal?

I'm guessing you are referring to [this](https://github.com/flathub/org.tlauncher.TLauncher/issues/9). The only person there actually on the Flatpak team is barthalion who removed it promptly. Jurf is the original (and sole contributor) and has no real stake in the administration of flatpak more than you or me. Everybody else appears to be randoms or possibly even trolls.

**Comment by jamierocks** [comment]
> removed it promptly

I don't think 17 days from reporting is prompt, or being generous - I don't think 3 days between comments is prompt either. Though the reporting isn't really the issue, its the fact that it in any way passed their review process initially.

I was hoping that the maintainers were unaware of the piracy considerations, but that wasn't the case - the following was made very clear in the original issue.

> There’s also the question of whether this belongs on Flathub, as it allows to run Minecraft without an official account. On the other side, Google Play doesn’t seem to have a problem with this.

Given how other members of the Flathub organisation backed down on piracy in that chain, I don't think publication to said service is something that ATLauncher should do themselves.

**Comment by nickavem** [comment]
> There’s also the question of whether this belongs on Flathub, as it allows to run Minecraft without an official account. On the other side, Google Play doesn’t seem to have a problem with this.

Where is this from?

I can also see exactly where they were coming from in that quote. They may not completely understand Minecraft's redistribution policy, and may have figured it was ok since the app was on a monolithic platform like google play. I once again have no intention of being antagonistic, but I am failing to see how exactly Flathub supported piracy... They seem to have just made a mistake.

**Comment by Doomsdayrs** [comment]
I fail to see how ATLauncher would be considered piracy.

**Comment by Doomsdayrs** [comment]
Speaking of automation, automation of what exactly?

**Comment by Doomsdayrs** [comment]
Wait a minute, the flatpak package is just the GDLauncher flatpak with little change.

In fact, it is just the GDLauncher script renamed.

**Comment by Doomsdayrs** [comment]
This has been on the drawing board for too long. I am making a PR to flathub

**Comment by Doomsdayrs** [comment]
Gotta go fast https://github.com/flathub/flathub/pull/2799

**Pull request closed without merging**